pennylane/measurements.py

```diff
@@ -803,6 +805,39 @@ def circuit():
     return MeasurementProcess(State, wires=wires, shape=shape, numeric_type=complex)
 
 
+def vn_entropy(wires):
+    r"""Von Neumann entropy of the system prior to measurement.
+
+    Args:
+        wires (Sequence[int] or int): the wires of the subsystem
+
+        **Example:**
+
+    .. code-block:: python3
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.IsingXX(x, wires=[0, 1])
+            return qml.vn_entropy(wires=[0])
+
+    Executing this QNode:
+
+    >>> circuit_entropy(np.pi/2)
+    0.6931472
+
+    .. note::
+
+        Calculating the derivative of :func:`~.vn_entropy` is currently only supported when
+        using the classical backpropagation differentiation method (``diff_method="backprop"``)
+        with a compatible device.
+    """
+    # pylint: disable=protected-access
+    wires = qml.wires.Wires(wires)
+    return MeasurementProcess(VnEntropy, wires=wires, numeric_type=float)
```

Comment:
Measurement processes should define a shape if not dependent on the device.
```suggestion
    return MeasurementProcess(VnEntropy, wires=wires, shape=(1,), numeric_type=float)
```